jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/response/ResponseHeaders.java

```diff
@@ -0,0 +1,82 @@
+package se.fortnox.reactivewizard.jaxrs.response;
+
+import rx.Observable;
+import rx.Observer;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Allows you to return response headers together with your result.
+ */
+public class ResponseHeaders {
```

Comment:
The name here indicates that it is the headers of a response, while it's actually a response that also contains headers. Could it be named simply Response or ResponseWithHeaders if something more verbose is wanted?

Reply:
The class is actually not holding the response, that is done by the private class ObservableWithHeaders.

The ResoponseHeaders class actually just has the withHeaders method, to do the decoration.

I am open to other names, but ResponseWithHeaders would be misleading.

Reply:
So the class is more of a ResponseDecorator?

Reply:
@tsfcode you nailed it!